include/libtorrent/alert_types.hpp

```diff
@@ -1232,8 +1245,6 @@ namespace libtorrent
 	// listen on it.
 	struct TORRENT_EXPORT listen_failed_alert final : alert
 	{
-		enum socket_type_t { tcp, tcp_ssl, udp, i2p, socks5, utp_ssl };
```

Comment:
you still need the old enum as deprecated, just like the old name, ``sock_type``